### PR TITLE
Enhancement: Rework GithubAction workflows

### PR DESCRIPTION
**Pull request by OskarStark**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -20,16 +20,18 @@ jobs:
             -   name: "Checkout"
                 uses: actions/checkout@v2
 
-            -   name: "Cache dependencies installed with composer"
-                uses: actions/cache@v1
+            -   name: "Install PHP with extensions"
+                uses: "shivammathur/setup-php@v1"
```
Comment (smoench) [inline .github/workflows/build.yaml:24]: What do you think about migrating setup-php action to v2 https://github.com/shivammathur/setup-php/wiki/Switch-to-v2?

Comment (OskarStark) [reply .github/workflows/build.yaml:24]: Yes we just need to change the version to v2 IMO

**Review by smoench — APPROVED** [review]
LGTM!

**Comment by smoench** [comment]
There are some warnings about using ubuntu-latest it will change from ubuntu-18.04 to ubuntu-20.04 soon

**Comment by OskarStark** [comment]
> There are some warnings about using ubuntu-latest it will change from ubuntu-18.04 to ubuntu-20.04 soon

Thanks, I think we can ignore them for now

**Pull request merged**